src/world/action/slide.c

```diff
@@ -1,5 +1,202 @@
+            sp48 = player_raycast_below_cam_relative(playerStatus, &sp28, &sp2C, &sp30, &sp34, &sp38, &sp3C, &sp40, &sp44);
+            if (sp48 >= 0) {
+                tempCurrentSpeed = playerStatus->currentSpeed / 3.0f;
+                if (tempCurrentSpeed < 0.01) {
+                    playerStatus->currentSpeed = 0.0f;
+                }
+                playerStatus->currentSpeed -= tempCurrentSpeed;
+                if (playerStatus->currentSpeed <= 0.0f) {
+                    sfx_play_sound_at_player(0x172, 0);
+                    suggest_player_anim_setUnkFlag(0x10031);
+                    playerStatus->fallState = 6;
+                    playerStatus->framesOnGround = 15;
+                    playerStatus->currentSpeed = 0.0f;
+                    playerStatus->position.y = sp2C;
+                    break;
+                }
+                break;
+            }
+        case 2:
+            playerStatus->currentSpeed += D_802B678C;
+            if (playerStatus->currentSpeed <= 0.0f) {
+                playerStatus->currentSpeed = 0.0f;
+            }
+            playerStatus->gravityIntegrator[0] += playerStatus->gravityIntegrator[1];
+            playerStatus->position.y += playerStatus->gravityIntegrator[0];
+            if (playerStatus->gravityIntegrator[0] <= 0.0f) {
+                playerStatus->fallState = 3;
+                playerStatus->gravityIntegrator[0] = 0.1143f;
+                playerStatus->gravityIntegrator[1] = -0.2871f;
+                playerStatus->gravityIntegrator[2] = -0.1823f;
+                playerStatus->gravityIntegrator[3] = 0.01152f;
+            }
+            break;
+        case 3:
+            playerStatus->currentSpeed += D_802B678C;
+            if (playerStatus->currentSpeed <= 0.0f) {
+                playerStatus->currentSpeed = 0.0f;
+            }
+            playerStatus->position.y = player_check_collision_below(func_800E34D8(), &sp48);
+            if (sp48 >= 0) {
+                D_802B678C = -1;
+                suggest_player_anim_setUnkFlag(0x80003);
+                sfx_play_sound_at_player(0x162, 0);
+                playerStatus->fallState++;
+                break;
+            }
+            break;
+        case 4:
+            playerStatus->currentSpeed += D_802B678C;
+            if (playerStatus->currentSpeed <= 0.0f) {
+                playerStatus->currentSpeed = 0.0f;
+            }
+            if (playerStatus->unk_BC != 0) {
+                suggest_player_anim_setUnkFlag(0x10030);
+                playerStatus->fallState++;
+            }
+            break;
+        case 5:
+            playerStatus->currentSpeed += D_802B678C;
+            if (playerStatus->currentSpeed <= 0.0f) {
+                playerStatus->currentSpeed = 0.0f;
+            }
+            if (playerStatus->unk_BC != 0) {
+                suggest_player_anim_setUnkFlag(0x10031);
+                sfx_play_sound_at_player(0x172, 0);
```

Comment:
Ideally we'd add 0x172 to enums.h's sound IDs listing; feel free to name it unk_172 or something.